app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,113 @@
 "use client";
 import PageLayout from "../../pageLayout/pageLayout";
-import React from "react";
-import SearchResults from "../../search/results";
+import React, { useState, useRef, useEffect } from "react";
 import {
   Box,
   Heading,
   HorizontalRule,
   SearchBar,
+  Menu,
+  Text,
+  Pagination,
 } from "@nypl/design-system-react-components";
+import { usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import {
+  displayResults,
+  totalNumPages,
+  createCollectionsQueryStringFromHash,
+} from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
+import NoResultsFound from "../../results/noResultsFound";
+import {
+  DEFAULT_PAGE_NUM,
+  DEFAULT_COLLECTION_SORT,
+  DEFAULT_SEARCH_TERM,
+} from "@/src/config/constants";
+
+export function CollectionsPage({ data, params, renderCollections }) {
+  const { push } = useRouter();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+  let collections = [];
+
+  if (renderCollections) {
+    collections = Array.isArray(data.collection)
+      ? data.collection
+      : [data.collection];
+  }
+
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  const numFound = data.numFound || data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  const [currentPage, setCurrentPage] = useState<number>(
+    Number(params.page) || Number(DEFAULT_PAGE_NUM)
+  );
+
+  const [currentSort, setCurrentSort] = useState<string>(
+    params.sort || DEFAULT_COLLECTION_SORT
+  );
+
+  const [currentCollectionKeywords, setcurrentCollectionKeywords] =
+    useState<string>(params.collection_keywords || DEFAULT_SEARCH_TERM);
+
+  const updateURL = async (queryString) => {
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}#collections`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 1000);
+  };
+
+  const handleSearchSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    // something is so weird about state here.
+    // change the values in the object passed down to createCollectionsQueryStringFromObject to be currentPage and currentSort and tell me if it works for you....
```

Comment:
Ah yeah, that's because the `set...` functions are asynchronous while the `createCollectionsQueryStringFromObject` is not. So this is updating the current values for page and sort, not waiting until that's done, and calling the function with the page and sort values. You might be seeing stale values _before_ they get updated from the async calls.

One option is to put these type of side effects inside the set function as
```
let queryString = "";
setCurrentPage((_prevValue) => {
  queryString = createCollectionsQueryStringFromHash({
      collection_keywords: currentCollectionKeywords,
      sort: DEFAULT_COLLECTION_SORT,
      page: DEFAULT_PAGE_NUM,
    });
  return Number(DEFAULT_PAGE_NUM);
}
```

This should work but it's only referencing the page value. With TWO update functions, this gets trickier. You can try doing a double nested callback function but, as you can see from the example above, it'll get messy...